### PR TITLE
Acoular demo + Import speed up

### LINKED ISSUE #130: Make acoular_demo.py matplotlib independent + call in tests
modify acoular_demo.py such that it can also be executed if matplotlib cannot be imported (e.g. print mic geom as array, calculate max or sum of array), add it to tests

### LINKED ISSUE #212: Use importlib instead of try and except
The availability of a certain module should be verified with the importlib module (without importing). 

See: https://docs.python.org/3/library/importlib.html#checking-if-a-module-can-be-imported

Changes are related to [configuration.py](acoular/configuration.py#L130-145)

### PR DESCRIPTION
**Pull request by adku1173**
* adds a test that `acoular_demo.py` runs as expected (fixes #130 ) 
* uses `importlib` to check if packages can be imported (without actually importing them) -> speeds up Acoular import by > 1s! (fixes #212 )

### CODE REVIEW

File: acoular/configuration.py [hunk 1]
```diff
@@ -117,6 +118,21 @@ def __init__(self):
...
     _use_traitsui = Bool(False)
 
+    #: Boolean Flag that determines whether tables is installed.
+    tables_exists = Property()
+
+    #: Boolean Flag that determines whether h5py is installed.
+    h5py_exists = Property()
+
+    #: Boolean Flag that determines whether matplotlib is installed.
+    matplotlib_exists = Property()
```
[the remark this answers is not in the document]
Comment (adku1173) [reply acoular/configuration.py:128]: We could do this, but for me, `foo_exists` sounds more natural, especially in the context of `if foo_exists` as opposed to `if have_foo`.

File: pyproject.toml [hunk 2]
```diff
@@ -94,6 +94,7 @@ python = ["3.7", "3.8", "3.9", "3.10", "3.11"]
 [tool.hatch.envs.tests.scripts]
 import = ["python -c \"import acoular\""]
 test = ["python -m pytest -v tests"]
+demo = ["python -c \"import acoular; acoular.demo.acoular_demo.run()\""]
```
[the remark this answers is not in the document]
Comment (adku1173) [reply pyproject.toml:97]: My initial code considers the usual way to call the acoular_demo (the way it is described in the Acoular docs). But I see that we then test the Acoular import twice. What about deleting the import test?

[the remark this answers is not in the document]
Comment (artpelling) [reply pyproject.toml:97]: The suggested way checks the executability of the file and does not import acoular. If you want this to be consistent we could change the docs... I don't really care tbh

File: acoular/configuration.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (adku1173) [reply acoular/configuration.py]: accepted

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (adku1173) [reply pyproject.toml:97]: I believe it is not enough to test the file's executability. The `acoular_demo` must be importable and executable.

**Pull request merged**